plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceProviderDescriptor.java

```diff
@@ -174,9 +174,18 @@ public DataSourceProviderDescriptor(DataSourceProviderRegistry registry, IConfig
 
         // Load native clients
         {
-            for (IConfigurationElement nativeClientsElement : config.getChildren("nativeClients")) {
-                for (IConfigurationElement clientElement : nativeClientsElement.getChildren("client")) {
-                    this.nativeClients.add(new NativeClientDescriptor(clientElement));
+            IConfigurationElement[] nativeClients = config.getChildren("nativeClients");
+            if (ArrayUtils.isEmpty(nativeClients)) {
```

Comment:
Why can't we have both `nativeClients` and `inheritClients` at the same time? IMO it makes sense.